policy-controller/k8s/api/src/policy/httproute.rs

```diff
@@ -161,3 +161,60 @@ pub struct HttpRouteStatus {
     #[serde(flatten)]
     pub inner: RouteStatus,
 }
+
+impl HttpRouteRule {
+    pub fn has_relative_paths(&self) -> bool {
+        matches_have_relative_paths(&self.matches)
+            || self
+                .filters
+                .iter()
+                .flatten()
+                .any(HttpRouteFilter::has_relative_paths)
+    }
+}
+
+impl HttpRouteFilter {
+    pub fn has_relative_paths(&self) -> bool {
+        match self {
+            HttpRouteFilter::RequestRedirect {
+                ref request_redirect,
+            } => request_redirect_has_relative_paths(request_redirect),
+            _ => false,
+        }
+    }
+}
+
+pub fn parent_ref_targets_server(p: &ParentReference) -> bool {
```

Comment:
If we're going to split this into a helper, it could take a generic resource type (as the Target types do).

Reply:
oh, you mean like this? https://github.com/linkerd/linkerd2/blob/c3d327f9c7cfe3c7ea85df8aaf89036388b63b07/policy-controller/k8s/api/src/policy/target_ref.rs#L192-L206

yeah, we could do that